avl_tree/AVL_Tree.java

```diff
@@ -0,0 +1,294 @@
+			if (node.right.hRight - node.right.hLeft >= 1)
+				node = rotateRR(node);
+			else if (node.right.hRight - node.right.hLeft <= -1)
+				node = rotateRL(node);
+		} else if (node.hRight - node.hLeft <= -2) {
+			if (node.left.hRight - node.left.hLeft <= -1)
+				node = rotateLL(node);
+			else if (node.left.hRight - node.left.hLeft >= 1)
+				node = rotateLR(node);
+		}
+		return node;
+	}
+	public boolean isEmpty(){
+		return root == null;
+	}
+	public void prntIn(NodeAVL node) {
+		if (node == null)
+			return;
+		else if (node.left == null && node.right == null)
+			System.out.print(node.data + " ");
+		else {
+			prntIn(node.left);
+			System.out.print(node.data + " ");
+			prntIn(node.right);
+		}
+	}
+
+	public void del(int data) {
+		if (root.data == data) {
+			NodeAVL temp = root.right;
+			if (root.left == null && root.right == null)
+				root = null;
+			else if (root.left == null)
+				root = root.right;
+			else if (temp == null)
+				root = root.left;
+			else {
+				int dta = 0;
+				if (root.right.left == null) {
+					root.right.left = root.left;
+					root = root.right;
+				} else {
+					dta = go(temp);
+					root.data = dta;
+				}
+				if (root.right == null)
+					root.hRight = 0;
+				else
+					root.hRight = Math.max(root.right.hLeft, root.right.hLeft);
+				// System.out.print(root.data);
+				root = isRotate(root);
+			}
+		} else
+			del(root, data);
+	}
+
+	private void del(NodeAVL node, int data) {
+		if (node.right == null && node.left == null){
+			System.out.println("Element you wanna delete not exist");
+			return;
+		}
+		if (node.right != null && node.right.data == data) {
+			NodeAVL del = node.right;
+			if (del.right == null && del.left == null)
+				node.right = null;
+			else if (del.left == null)
+				node.right = del.right;
+			else if (del.right == null)
+				node.right = del.left;
+			else {
+				NodeAVL temp = del.right;
+				if (temp.left == null)
+					node.right = node.right.right;
+				else
+					del.data = go(temp);
+				del.hRight = Math.max(del.right.hLeft, del.right.hRight);
+			}
+		} else if (node.left != null && node.left.data == data) {
+			NodeAVL del = node.left;
+			if (del.right == null && del.left == null)
+				node.left = null;
+			else if (del.left == null)
+				node.left = del.right;
+			else if (del.right == null)
+				node.left = del.left;
+			else {
+				NodeAVL temp = del.right;
+				if (temp.left == null)
+					node.left = node.left.right;
+				del.data = go(temp);
+				del.hRight = Math.max(del.right.hLeft, del.right.hRight);
+			}
+		} else if (node.data > data) {
+			del(node.left, data);
+			if (node.left == null)
+				node.hLeft = 0;
+			else
+				node.hLeft = Math.max(node.left.hLeft, node.left.hRight) + 1;
+		} else if (node.data < data) {
+			del(node.right, data);
+			if (node.right == null)
+				node.hRight = 0;
+			else
+				node.hRight = Math.max(node.right.hLeft, node.right.hRight) + 1;
+		}
+		node = isRotate(node);
+	}
+
+	public int go(NodeAVL node) {

```

Comment:
@Hsankesara same goes here, `informative name` not `informative comments`, also make it a little more explanatory where will this be used?